### PR TITLE
feat(querier): propagate all known alerting headers

### PR DESCRIPTION
**Pull request by JohnnyQQQQ**
Adding a few headers we missed in the first iteration, and making the code a bit easier to follow.

### CODE REVIEW

File: pkg/registry/apis/query/query.go [hunk 1]
```diff
@@ -135,24 +135,25 @@ func (r *queryREST) Connect(connectCtx context.Context, name string, _ runtime.O
...
-			"content-type":   "Content-Type",
-			"content-length": "Content-Length",
-			"user-agent":     "User-Agent",
-			"accept":         "Accept",
+		// Set of headers that we want to forward to the datasource api servers. Those are used i.e. for
+		// cache control or identifying the source of the request.
+		//
+		// The headers related to grafana alerting can be found here:
+		// https://github.com/grafana/grafana-ruler/blob/96e6d4b25c0d973a7615b92b35739511a6fbd72f/pkg/ruler/rulesmanager/ds_query_rule_evaluator.go#L313-L328
+		expectedHeaders := map[string]struct{}{
```
Comment (JohnnyQQQQ) [inline pkg/registry/apis/query/query.go:143]: If we want to indicate a set in go, this is usually how we would do it.

File: pkg/registry/apis/query/query.go [hunk 2]
```diff
@@ -39,6 +40,22 @@ var (
...
 	_ rest.StorageMetadata      = (*queryREST)(nil)
+	// Set of headers that we want to forward to the datasource api servers. Those are used i.e. for
+	// cache control or identifying the source of the request.
+	//
+	// The headers related to grafana alerting can be found here:
+	// https://github.com/grafana/grafana-ruler/blob/96e6d4b25c0d973a7615b92b35739511a6fbd72f/pkg/ruler/rulesmanager/ds_query_rule_evaluator.go#L313-L328
+	//
+	// The usage of strings.ToLower is because the server would convert `FromAlert` to `Fromalert`. So the make matching
+	// easier, we just match all headers in lower case.
+	expectedHeaders = map[string]string{
```
Comment (JohnnyQQQQ) [inline pkg/registry/apis/query/query.go:51]: Only init it once, as it's in the critical call path and to reduce allocations.

**Review by sarahzinger — APPROVED** [review]
Thanks for catching all that! Do we want to add these to the subquery requests in the datasource apiservers as well? https://github.com/grafana/grafana/pull/92304

**Comment by JohnnyQQQQ** [comment]
> Thanks for catching all that! Do we want to add these to the subquery requests in the datasource apiservers as well? #92304

Good catch! Added it also for `sub_query.go`

**Review by sarahzinger — APPROVED** [review]
nice thank you!

**Pull request merged**